__init__.py

```diff
@@ -27,13 +27,33 @@ def default(self, obj):
             return list(obj)
         return json.JSONEncoder.default(self, obj)
 
+class DistroImageValidation(click.Option):
+    def __init__(self, *args, **kwargs):
+        self.required_if_distro:list = kwargs.pop("required_if_distro")
+
+        assert self.required_if_distro, "'required_if_distro' parameter required"
+        kwargs["help"] = (kwargs.get("help", "") + "Option is required when distro is " + ", ".join(self.required_if_distro) + ".").strip()
+        super(DistroImageValidation, self).__init__(*args, **kwargs)
+
+    def handle_parse_result(self, ctx, opts, args):
+        current_opt:bool = self.name in opts
+        tgt_distro:str = opts["distro"]
+        for distro_opt in self.required_if_distro:
+            if distro_opt.casefold() == tgt_distro.casefold():
+                if current_opt:
+                    self.prompt = None
+                else:
+                    raise click.UsageError("Missing argument: '" + str(self.name) + "' is required with " + str(distro_opt) + " distro.")
+        return super(DistroImageValidation, self).handle_parse_result(ctx, opts, args)
+
 @click.command()
-@click.option('--distro', type=click.Choice(sorted(DISTROS.keys()) + ['*'], case_sensitive=False))
+@click.option('--distro', type=click.Choice(sorted(list(DISTROS.keys())) + ['*'], case_sensitive=False))
 @click.option('--version', required=False, default='')
 @click.option('--arch', required=False, type=click.Choice(['x86_64', 'aarch64'], case_sensitive=False), default='x86_64')
 @click.option('--out_fmt', required=False, type=click.Choice(['plain', 'json', 'driverkit'], case_sensitive=False),  default='plain')
-def crawl(distro, version='', arch='', out_fmt=''):
-    res = crawl_kernels(distro, version, arch, out_fmt == 'driverkit')
+@click.option('--image', cls=DistroImageValidation, required_if_distro=["Redhat"], multiple=True)
```

Comment:
I don't get how does this work:
* you can pass multiple image but only for Redhat, so if you use `*`, and you pass multiple images, what does happen?
* Then, crawl_kernels() seems to only care about a single image, right?

Reply:
I wrote this only for Redhat as this time, it is the only distro to require an image; the others can work from the mirrors. I thought from your previous comment about multiple images you meant building the various versions of Redhat from one call but I see now that it could have meant to actually have different images for different distros 🤔 So, at the moment, you can pass multiple images and it will work with multiple images but they can only be Redhat images (e.g. build, 6, 7, 8 and 9 in one command).

I can change the code so that it maybe extracts the `ID` from '/etc/*-release` and then the target would be that value rather than something passed in via command line (I can't think of any other way when there can be multiple targets and multiple images to make them match up). What do you think?

Reply:
I think it is ok as-is :) 
If i understood correctly, adding another container-based distro is just a matter of:
* adding new distro to `required_if_distro` list
* adding the new $DistroContainer to  DISTROS
* of course, adding the implementation of $DistroContainer
right?

Reply:
yup, exactly this (although I like the idea also of pulling target name from the container and using that too - i'll have a think about that in the background) 👍

Reply:
I am not sure we'd want this kind of automatic wizardry in such a kiss project :) 
But i am open to review such a PR!